### PR TITLE
Support auto_incr per CN cache.

### PR DESCRIPTION
**Pull request by iceTTTT**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6956 #7543 

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/frontend/types.go [hunk 1]
```diff
@@ -104,6 +104,11 @@ const (
 	externSql       = "external_sql"
 )
 
+// Cache size of auto_increment_columns cache.
+const (
+	cacheSize = 1000
```
Comment (ouyuanning) [inline pkg/frontend/types.go:109]: 1000 is too small

File: pkg/frontend/routine_manager.go [hunk 2]
```diff
@@ -37,6 +40,79 @@ type RoutineManager struct {
...
+	defer rm.mu.Unlock()
+	_, ok := rm.autocaches[oldname]
+	if ok {
+		rm.autocaches[newname] = AutoIncrCache{rm.autocaches[oldname].curNum,
+			rm.autocaches[oldname].maxNum, rm.autocaches[oldname].step}
+		delete(rm.autocaches, oldname)
+	}
+}
+
+func (rm *RoutineManager) GetNextCacheId(colDefs []*plan.ColDef, ctx context.Context, param *colexec.AutoIncrParam, bat *batch.Batch, tableID uint64) ([]uint64, []uint64, error) {
```
Comment (ouyuanning) [inline pkg/frontend/routine_manager.go:72]: maybe call   GetNextAutoIncrNum  is better

File: pkg/frontend/routine_manager.go [hunk 3]
```diff
@@ -37,6 +40,79 @@ type RoutineManager struct {
...
+	rm.mu.Lock()
+	defer rm.mu.Unlock()
+	offset, step := make([]uint64, 0), make([]uint64, 0)
+	for i, col := range colDefs {
+		if !col.Typ.AutoIncr {
+			continue
+		}
+
+		name := fmt.Sprintf("%d_%s", tableID, col.Name)
```
Comment (ouyuanning) [inline pkg/frontend/routine_manager.go:81]: i do not know if only use tableID for key is better?

**Review by ouyuanning — COMMENTED** [review]
maybe you can make a meeting to share the design

File: pkg/vm/process/types.go [hunk 4]
```diff
@@ -94,6 +94,7 @@ type SessionInfo struct {
 	StorageEngine  engine.Engine
 	QueryId        []string
 	ResultColTypes []types.Type
+	Ses            any
```
Comment (reusee) [inline pkg/vm/process/types.go:97]: I suggest using a full word but not an abbreviation, too hard to understand.

**Review by reusee — CHANGES_REQUESTED** [review]

**Review by reusee — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by nnsgmsone — COMMENTED** [review]
If one insertion acts on both cn again. This cache how to ensure the correctness of it.

**Review by nnsgmsone — APPROVED** [review]

**Review by aptend — APPROVED** [review]

**Pull request merged**